### DISCUSSION TITLE
What is the best way to search for a hash?

### DISCUSSION THREAD

**Discussion by laxas** [lancedb/lancedb#1894] [Q&A]
I got sha256 hashes (hexdigest) of the text chunks as a column in my table. To check if a text is allready in the table I need to find the hash of a probably new chunk. 

I tried 

```table.search().select(["sha"]).where("sha='xxx'").to_arrow()```
 
which takes about 250ms.

Is there a better way? Is it possible to build an index that speeds it up?

**Comment by albertlockett** [comment]
Hi @laxas . Do you have an index on the `sha` column? If not, it might be a good idea to create one as it could significantly improve query performance. 

It looks like you're using the synchronous python SDK, so you could try something like:
```py
table.create_scalar_index("sha", index_type="BTREE")
```

Or if you're using the async python SDK, it would be:
```py
from lancedb.index import BTree

await table.create_index("sha", config=BTree())
```
